WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionDeeplinkActivity.java

```diff
@@ -78,6 +78,7 @@ protected void onCreate(Bundle savedInstanceState) {
             // if user is signed in wpcom show the post right away - otherwise show welcome activity
             // and then show the post once the user has signed in
             if (mAccountStore.hasAccessToken()) {
+                trackSuccess(source);
                 if (source == Source.STATS) {
```

Comment:
This code can be simplified and improved a bit if we extract the following into a method:

```
if (source == Source.STATS) {
    showStats();
} else {
    finish();
}
```
Especially the `if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK && source == Source.STATS)` can be a little tricky, because the `else` statement might just handle a different result code, or a different event and just `finish()` the activity. So, something like this might work better:

```
private void trackAndFinishActivity() {
    JetpackConnectionWebViewActivity.trackWithSource(AnalyticsTracker.Stat.SIGNED_INTO_JETPACK, source);
    if (source == Source.STATS) {
        showStats();
    } else {
        finish();
    }
}

@Override
protected void onActivityResult(int requestCode, int resultCode, Intent data) {
    super.onActivityResult(requestCode, resultCode, data);
    if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK) {
        trackAndFinishActivity();
    }
}
```
_The method name could be something better_